### PR TITLE
remove: remove dvc.yaml and dvc.lock if they are empty

### LINKED ISSUE #4090: remove: remove dvc.yaml and dvc.lock if they are empty
https://github.com/iterative/dvc/pull/4074#issuecomment-648097445

```
$ cat dvc.lock
{}                                               
$ cat dvc.yaml
stages: {}                                       
```

### PR DESCRIPTION
**Pull request by nik123**
Fixes #4090 

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

https://github.com/iterative/dvc.org/issues/1564

### CODE REVIEW

File: dvc/dvcfile.py [hunk 1]
```diff
@@ -234,6 +234,8 @@ def remove_stage(self, stage):
         logger.debug("Removing '%s' from '%s'", stage.name, self.path)
         del d["stages"][stage.name]
         dump_yaml(self.path, d)
+        if len(d["stages"]) == 0:
+            self.remove(force=True)
```
Comment (skshetry) [inline dvc/dvcfile.py:238]: Let's not dump if there's going to be no stages at all.

Comment (skshetry) [reply dvc/dvcfile.py:238]: Also, I think it's wrong to assume that lock file and dvc.yaml will always have the same no. of entries.

File: tests/func/test_dvcfile.py [hunk 2]
```diff
@@ -285,6 +285,32 @@ def test_remove_stage_preserves_comment(tmp_dir, dvc, run_copy):
...
+    tmp_dir.gen(
+        "dvc.yaml",
+        textwrap.dedent(
+            """\
+            stages:
+                generate-foo:
+                    cmd: "echo foo > foo"
+                    outs:
+                    - foo"""
+        ),
```
Comment (skshetry) [inline tests/func/test_dvcfile.py:298]: @nik123, we don't care about yaml structure here. It'd be better either to create stage via `run_copy` fixture or use `dump_yaml`.

File: dvc/dvcfile.py (hunk 1 above)
Comment (nik123) [reply dvc/dvcfile.py]: > Let's not dump if there's going to be no stages at all.

Fixed

Comment (nik123) [reply dvc/dvcfile.py]: > Also, I think it's wrong to assume that lock file and dvc.yaml will always have the same no. of entries. I hope they'd be the same, but at the end, they are two different files (one of which, you're allowed to edit/remove yourselves).

I'm not sure I've got you right but now `dvc.lock` and `dvc.yaml` are removed independently. Please, check it out.

File: tests/func/test_dvcfile.py (hunk 2 above)
Comment (nik123) [reply tests/func/test_dvcfile.py]: Fixed

**Pull request merged**